From 40235c5864019f8d5c1ca02e9fd2228aeb1e804f Mon Sep 17 00:00:00 2001
From: Wanhui Sun <wanhui.sun@intel.com>
Date: Mon, 11 Mar 2019 19:13:56 +0800
Subject: [PATCH 13/81] media: Add poll support

Implement poll for events. POLLPRI is used to notify users on incoming
events.

disable the poll function for build error.

Change-Id: I725cb31829256cde0b13997dd4fd820a208049a6
Signed-off-by: Chen Meng J <meng.j.chen@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
Signed-off-by: lijuncax <lijunx.cai@intel.com>
---
 drivers/media/mc/mc-device.c | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/drivers/media/mc/mc-device.c b/drivers/media/mc/mc-device.c
index 1c0e0e081fe5..d254ef19ddc7 100644
--- a/drivers/media/mc/mc-device.c
+++ b/drivers/media/mc/mc-device.c
@@ -501,6 +501,39 @@ out_free:
 	return ret;
 }
 
+static unsigned int media_device_poll(struct file *filp,
+                                     struct poll_table_struct *wait)
+{
+	/*
+	struct media_device_fh *fh = media_device_fh(filp);
+	struct media_device *mdev = fh->fh.devnode->media_dev;
+	unsigned int poll_events = poll_requested_events(wait);
+	int ret = 0;
+
+	if (poll_events & (POLLIN | POLLOUT))
+	        return POLLERR;
+
+	if (poll_events & POLLPRI) {
+	        unsigned long flags;
+	        bool empty;
+
+	        spin_lock_irqsave(&mdev->req_lock, flags);
+	        empty = list_empty(&fh->kevents.head);
+	        spin_unlock_irqrestore(&mdev->req_lock, flags);
+
+	        if (empty)
+	                poll_wait(filp, &fh->kevents.wait, wait);
+	        else
+	                ret |= POLLPRI;
+	}
+
+	return ret;
+	*/
+
+	printk("media_device_poll func\n");
+	return 0;
+}
+
 #ifdef CONFIG_COMPAT
 
 struct media_links_enum32 {
@@ -566,6 +599,7 @@ static const struct media_file_operations media_device_fops = {
 	.owner = THIS_MODULE,
 	.open = media_device_open,
 	.ioctl = media_device_ioctl,
+	.poll = media_device_poll,
 #ifdef CONFIG_COMPAT
 	.compat_ioctl = media_device_compat_ioctl,
 #endif /* CONFIG_COMPAT */
-- 
2.17.1

